pallets/cash/src/internal/lock.rs

```diff
@@ -0,0 +1,199 @@
+// Note: The substrate build requires these be re-exported.
+use frame_support::storage::{StorageMap, StorageValue};
+
+// Import these traits so we can interact with the substrate storage modules.
+use crate::{
+    chains::ChainAccount,
+    core::{repay_and_supply_principal, sub_principal_amounts},
+    reason::Reason,
+    types::{CashIndex, CashPrincipalAmount},
+    CashPrincipals, ChainCashPrincipals, Config, Event, GlobalCashIndex, Module,
+    TotalCashPrincipal,
+};
+
+pub fn lock_cash_principal_internal<T: Config>(
```

Comment:
note to other reviewers, this function diff is really something like this afaict 
![image](https://user-images.githubusercontent.com/3612857/110028577-694c9d00-7d01-11eb-8f61-d0f3851a5349.png)